yb-voyager/cmd/exportSchemaOptimizationReport.go

```diff
@@ -153,119 +165,109 @@ type SecondaryIndexToRangeChange struct {
 	redundantIndexes := indexTransformer.RemovedRedundantIndexes
 	if skipPerfOptimizations {
 		redundantIndexes = redundantIndexesToRemove
-		redundantIndexChange.Description = REDUNDANT_INDEXES_NOT_APPLIED_DESCRIPTION
-		redundantIndexChange.Title = "Redundant Indexes - Not Removed"
-		redundantIndexChange.IsApplied = false
 	}
-	redundantIndexChange.TableToRemovedIndexesMap = GetTableToIndexMap(redundantIndexes)
-	return redundantIndexChange
+	return NewRedundantIndexChange(!bool(skipPerfOptimizations), redundantIndexesFile, RedundantIndexesFileName, GetTableToIndexMap(redundantIndexes))
 }
 
 func buildShardingTableRecommendationChange(shardedTables []string, colocatedTables []string) *ShardingRecommendationChange {
-	var appliedRecommendationTable *ShardingRecommendationChange
 	if !assessmentRecommendationsApplied { //If assessment recommendations not applied and skip recommendations is true, then show that its not applied
 		if skipRecommendations {
-			appliedRecommendationTable = NewAppliedShardingRecommendationChange("") // Dummy entry for both table and mview as no need to show two
-			appliedRecommendationTable.IsApplied = false
-			appliedRecommendationTable.Description = APPLIED_RECOMMENDATIONS_NOT_APPLIED_DESCRIPTION
-			appliedRecommendationTable.Title = "Sharding Recommendations - Not Applied"
+			return NewAppliedShardingRecommendationChange("", false) // Dummy entry for both table and mview as no need to show two
 		}
-		return appliedRecommendationTable
+		return nil
 	}
 
-	tableFile := utils.GetObjectFilePath(filepath.Join(exportDir, "schema"), TABLE)
+	referenceTableFile := utils.GetObjectFilePath(filepath.Join(exportDir, "schema"), TABLE)
 	//To tables then add that change
-	if !utils.FileOrFolderExists(tableFile) || len(shardedTables) == 0 { // only display this in case there is any modifield sharded tables
+	if !utils.FileOrFolderExists(referenceTableFile) || len(shardedTables) == 0 { // only display this in case there is any modifield sharded tables
 		return nil
 	}
-	appliedRecommendationTable = NewAppliedShardingRecommendationChange(TABLE)
 	// Get relative path from reports directory to the table file
 	reportsDir := filepath.Join(exportDir, "reports")
-	relativePath, err := filepath.Rel(reportsDir, tableFile)
-	if err != nil {
-		// Fallback to absolute path if relative path calculation fails
-		appliedRecommendationTable.ReferenceFile = tableFile
-	} else {
-		appliedRecommendationTable.ReferenceFile = relativePath
+	relativePath, err := filepath.Rel(reportsDir, referenceTableFile)
+	if err == nil {
+		//If no error, then use the relative path
+		referenceTableFile = relativePath
 	}
-	appliedRecommendationTable.ReferenceFileDisplayName = filepath.Base(tableFile)
-	appliedRecommendationTable.ShardedObjects = shardedTables
-	appliedRecommendationTable.CollocatedObjects = colocatedTables
 
-	return appliedRecommendationTable
+	change := NewAppliedShardingRecommendationChange(TABLE, true)
```

Comment:
nit: again here, can we not pass all of these into the constructor, rather than setting them outside?